pkg/option/config.go

```diff
@@ -755,6 +760,18 @@ const (
 
 	// NodePortMaxDefault is the maximum port to listen for NodePort requests
 	NodePortMaxDefault = 32767
+
+	// KubeProxyReplacementProbe specifies to detect and enable available
+	// kube-proxy replacement features
+	KubeProxyReplacementProbe = "probe"
+
+	// KubeProxyReplacementStrict specifies to strictly enable selected
+	// kube-proxy replacement features
+	KubeProxyReplacementStrict = "strict"
+
+	// KubeProxyReplacementPartial specifies to enable only selected kube-proxy
+	// replacement features
+	KubeProxyReplacementPartial = "partial"
```

Comment:
```
type KubeProxyReplacementMode string

const (
        KubeProxyReplacementProbe KubeProxyReplacementMode = "probe"
        KubeProxyReplacementStrict  KubeProxyReplacementMode = "strict"
        KubeProxyReplacementPartial KubeProxyReplacementMode = "partial"
)
```

Reply:
Not sure whether an extra type would give us any benefits besides adding more code. Also, it's not consistent with existing flags, e.g. `--tunnel`.